api_client/python/timesketch_api_client/client.py

```diff
@@ -365,102 +367,232 @@ def _create_session(
-            RuntimeError: If the API server returns an error or empty.
+            requests.exceptions.ConnectionError: If a connection error persists
+                after all retry attempts.
+            ValueError: If the API response cannot be JSON-decoded after all
+                retry attempts.
+            RuntimeError: If the API server returns an error (non-20x status code)
+                or a "falsy" JSON response (e.g., null, empty list/dict)
+                after all retry attempts.
         """
         resource_url = "{0:s}/{1:s}".format(self.api_root, resource_uri)
 
-        retry_count = 0
+        # Start attempt count from 0, first loop with set it to 1
+        attempt = 0
         result = None
         while True:
-            retry_count += 1
-            response = self.session.get(resource_url, params=params)
+            attempt += 1
+            # If this is not the first attempt, wait before trying again
+            if attempt > 1:
+                # Simple fixed backoff before retrying
+                time.sleep(1)  # Wait for 1 second before the next attempt
+
             try:
+                response = self.session.get(resource_url, params=params)
                 result = error.get_response_json(response, logger)
                 if result:
                     return result
             except RuntimeError as e:
-                if retry_count >= self.DEFAULT_RETRY_COUNT:
+                if attempt >= self.DEFAULT_RETRY_COUNT:
+                    # Re-raise the original error after exhausting retries
                     raise RuntimeError(
                         "Error for request '{0:s}' - '{1!s}'".format(resource_url, e)
                     ) from e
 
                 logger.warning(
-                    "[{0:d}/{1:d}] Parsing the response for request '{2:s}'"
-                    "failed. Trying again...".format(
-                        retry_count, self.DEFAULT_RETRY_COUNT, resource_url
-                    )
+                    "[%d/%d] API error (RuntimeError) for request '%s' "
+                    "failed. Error: %s. Trying again...",
+                    attempt,
+                    self.DEFAULT_RETRY_COUNT,
+                    resource_url,
+                    e,
                 )
             except ValueError as e:
-                if retry_count >= self.DEFAULT_RETRY_COUNT:
-                    raise ValueError(
-                        "Error parsing response for request '{0:s}' - {1!s}".format(
-                            resource_url, e
-                        )
-                    ) from e
+                if attempt >= self.DEFAULT_RETRY_COUNT:
+                    # Re-raise the original error after exhausting retries
+                    error_msg = (
+                        f"Error parsing response for request '{resource_url}'"
+                        f" - {e!s}"
+                    )
+                    raise ValueError(error_msg) from e
```

Comment:
What do we expect to change with a retry that would fix a "ValueError"?

Reply:
Good question, the value in question here is something from the server, that is expected to be JSON, but due to several reasons, including network error or temporary server issues, the server COULD return not valid JSON. So while in theory, that should never be the case, it could, does that make sense?